core/src/main/scala/pureconfig/DurationConvert.scala

```diff
@@ -11,13 +15,8 @@ private[pureconfig] object DurationConvert {
   /**
    * Convert a string to a Duration while trying to maintain compatibility with Typesafe's abbreviations.
    */
-  def fromString[D](durationString: String, ct: ClassTag[D]): Try[Duration] = {
-    Try { Duration(addZeroUnit(justAMinute(itsGreekToMe(durationString)))) }
-      .recoverWith {
-        case ex: NumberFormatException =>
-          val err = s"Could not parse a ${ct.runtimeClass.getSimpleName} from '$durationString'. (try ns, us, ms, s, m, h, d)"
-          new util.Failure(new IllegalArgumentException(err, ex))
-      }
+  val fromString: String => Either[ConfigReaderFailure, Duration] = {
+    catchReadError(string => Duration(addZeroUnit(justAMinute(itsGreekToMe(string)))))
```

Comment:
Could we preserve the improved error message in this case?
I think something like this would do it:
```
catchReadError{ string =>
  Try(Duration(addZeroUnit(justAMinute(itsGreekToMe(string)))))
  .recoverWith {		
    case ex: NumberFormatException =>		
      val err = s"Could not parse a ${ct.runtimeClass.getSimpleName} from '$durationString'. (try ns, us, ms, s, m, h, d)"		
     new util.Failure(new IllegalArgumentException(err, ex))		
  }.get  
}